golem-rib/src/type_inference/enum_resolution.rs

```diff
@@ -71,14 +71,17 @@ mod internal {
                     inferred_type,
                     ..
                 } => {
-                    // Retrieve the possible no-arg variant from the registry
-                    let key = RegistryKey::FunctionName(variable_id.name().clone());
-                    if let Some(RegistryValue::Value(AnalysedType::Enum(typed_enum))) =
-                        function_type_registry.types.get(&key)
-                    {
-                        enum_cases.push(variable_id.name());
-                        *inferred_type = inferred_type
-                            .merge(AnalysedType::Enum(typed_enum.clone()).clone().into());
+                    // If variable is local, it takes priority over being a global enum
+                    if !variable_id.is_local() {
```

Comment:
This is the fix for enum - if the variable id is not local, then we go ahead and assign the type